pkg/kubecost/totals.go

```diff
@@ -0,0 +1,367 @@
+package kubecost
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/kubecost/cost-model/pkg/log"
+	"github.com/patrickmn/go-cache"
+)
+
+// TODO can we use ResourceTotals for all things or do we need specific structs
+// like AllocationResourceTotals, AssetResourceTotals, etc.
+
+type ResourceTotals struct {
+	Start                 time.Time `json:"end"`
+	End                   time.Time `json:"start"`
+	Cluster               string    `json:"cluster,omitempty"`
+	Node                  string    `json:"node,omitempty"`
```

Comment:
I think its worth briefly explaining why only these two fields are necessary to identify resource totals.

Reply:
Also, why are these `omitempty`?

Reply:
Frankly, we don't need struct tags at all. I don't serialize these anywhere. I'll just remove.

And I'm not opposed to leaving a comment about Cluster and Node, but it may not ultimately be true that those are the only two that are necessary. It's just that for now, that's all we ever use (e.g. idle by cluster or idle by node). So I will include that in the comment, but we may end up adding more properties later. Who knows. It's a kind of "fluid" type.